Pre prod v15

ashishverma8848 opened the pull request:
<!--

Some key notes before you open a PR:

 1. Select which branch should this PR be merged in?
 2. PR name follows [convention](http://karma-runner.github.io/4.0/dev/git-commit-msg.html)
 3. All tests pass locally, UI and Unit tests
 4. All business logic and validations must be on the server-side
 5. Update necessary Documentation
 6. Put `closes #XXXX` in your comment to auto-close the issue that your PR fixes


Also, if you're new here

- Documentation Guidelines => https://github.com/frappe/erpnext/wiki/Updating-Documentation

- Contribution Guide => https://github.com/frappe/erpnext/blob/develop/.github/CONTRIBUTING.md

- Pull Request Checklist => https://github.com/frappe/erpnext/wiki/Pull-Request-Checklist

-->

> Please provide enough information so that others can review your pull request:

<!-- You can skip this if you're fixing a typo or updating existing documentation -->

> Explain the **details** for making this change. What existing problem does the pull request solve?

<!-- Example: When "Adding a function to do X", explain why it is necessary to have a way to do X. -->

> Screenshots/GIFs

<!-- Add images/recordings to better visualize the change: expected/current behviour -->

Commit:
fix: erpnext commit on 21 oct

Commit:
fix: erpnext commit on 22 oct

Commit:
fix: erpnext commit on 23 oct

Commit:
fix: commits in erpnext on 23,24 and 25 oct

Commit:
fix: commits in erpnext on 26 oct

Commit:
fix: commits in erpnext on 28 oct

Commit:
fix: commits in erpnext on 29 oct

Commit:
fix: commints on erpnext on date 30 oct

Commit:
fix: commints on erpnext on date 1 and 3 nov

Commit:
fix: commints on erpnext on date 4 and 5 nov

Commit:
fix: commints on erpnext on date 6 nov

Commit:
fix: commints on erpnext on date 7,8 and 9 nov

ashishverma8848 committed:
Merge branch 'pre-prod' into pre_prod_v15

tinadn commented on erpnext/accounts/doctype/purchase_invoice/test_purchase_invoice.py:
```diff
@@ -406,17 +407,11 @@ def test_gl_entries_for_non_stock_items_with_perpetual_inventory(self):
+
+		expected_values = [
+			["_Test Account Cost for Goods Sold - TCP1", 250.0, 0],
+			["Creditors - TCP1", 0, 250],
```
Revert this change

tinadn commented on erpnext/accounts/doctype/purchase_invoice/test_purchase_invoice.py:
```diff
@@ -1324,10 +1385,8 @@ def test_gain_loss_with_advance_entry(self):
-			expected_gle = [["Cash - _TC", -70000.0],["_Test Payable USD - _TC", 70000.0]]
-		else:
-			expected_gle = [["_Test Payable USD - _TC", 70000.0], ["Cash - _TC", -70000.0]]
+
```
Revert this.

tinadn commented on erpnext/accounts/doctype/purchase_invoice/test_purchase_invoice.py:
```diff
@@ -1461,10 +1520,7 @@ def test_purchase_invoice_advance_taxes(self):
-			expected_gle = [["Creditors - _TC", -30000],["_Test Account Cost for Goods Sold - _TC", 30000]]
-		else:
-			expected_gle = [["_Test Account Cost for Goods Sold - _TC", 30000], ["Creditors - _TC", -30000]]
+		expected_gle = [["_Test Account Cost for Goods Sold - _TC", 30000], ["Creditors - _TC", -30000]]
```
Revert this.

tinadn commented on erpnext/accounts/doctype/purchase_invoice/test_purchase_invoice.py:
```diff
@@ -1942,17 +2053,10 @@ def test_repost_accounting_entries(self):
 		)
-		if frappe.db.db_type=='postgres':
-			expected_gle = [
+		expected_gle = [
```
Revert this.

tinadn commented on erpnext/accounts/doctype/purchase_invoice/test_purchase_invoice.py:
```diff
@@ -1913,10 +2024,10 @@ def test_offsetting_entries_for_accounting_dimensions(self):
 		pi.submit()
 
 		expected_gle = [
+			["_Test Account Cost for Goods Sold - _TC", 1000, 0.0, nowdate(), branch2.branch],
```
Revert this

tinadn commented on erpnext/controllers/sales_and_purchase_return.py:
```diff
@@ -276,39 +290,38 @@ def get_returned_qty_map_for_row(return_against, party, row_name, doctype):
-			Sum(Abs(child.received_qty)).as_("received_qty")
-		)
+		fields += [
+			f"sum(abs(`tab{child_doctype}`.rejected_qty)) as rejected_qty",
```
Revert from line no. 293 to 324 as we face issue in frappe.get_all()

Commit:
fix: revert changes in file test_purchase_invoice.py

Commit:
Merge branch 'pre_prod_v15' of https://github.com/8848digital/erpnext into pre_prod_v15

ashishverma8848 committed:
Update test_purchase_invoice.py

revert the changes

ashishverma8848 committed:
Update sales_and_purchase_return.py

revert changes in function def get_returned_qty_map_for_row(return_against, party, row_name, doctype):

ashishverma8848 committed:
Update general_ledger.py

change in def validate_disable_account
